src/modules/configs/config.py

```diff
@@ -81,3 +81,120 @@ class OperatorConf(Conf):
+                    "Order is undefined. Rule mode order needs to be defined for rule mode.")
+            if not value.get(mode):
+                raise Exception("Mode is undefined")
+            final_conf[key] = self.get_final_value(key)
+
+        return final_conf
+
+    def get_final_value(self, merged_key):
+        value = self.apply_config_rule(merged_key)
+        if value is None:
+            return CONFIG_RULES[merged_key].get("default")
+
+        return value
+
+    def apply_config_rule(self, merged_key):
+        mode = CONFIG_RULES.get(merged_key).get("mode")
+        if mode == "exclusive":
+            value = self.apply_exclusive_rule(merged_key)
+        elif mode == "inclusive":
+            value = self.apply_inclusive_rule(merged_key)
+        elif mode == "priority":
+            value = self.apply_priority_rule(merged_key)
+        else:
+            raise Exception("Unexpected mode found, review CONFIG_RULES")
+
+        return value
+
+    def merge_conf(self, dct):
+        merged_conf = {}
+        for key, value in dct.items():
+            if isinstance(value, dict):
+                nested = self.merge_conf(value)
+                for nested_key, nested_value in nested.items():
+                    merged_conf[f"{key}_{nested_key}"] = nested_value
+            else:
+                merged_conf[key] = value
+
+        return merged_conf
+
+    def apply_exclusive_rule(self, merged_key):
+        """
+        1. Check if value is defined in both -> raise Exc
+        2. Return value if defined in any
+        3. If undefined -> return None
+        """
+        order = CONFIG_RULES.get(merged_key)["exclusive"]
+
+        value_count = 0
+        value, effective_value = None, None
+        for config in order:
+            value = self.merged_conf[config].get(merged_key)
+            if value:
+                value_count += 1
+                effective_value = value
+            if value_count > 1:
+                raise Exception(f"{merged_key} is defined in more than one config.")
+        if value_count == 0:
+            raise Exception(f"{merged_key} is not defined in any config.")
+        return effective_value
+
+    def apply_inclusive_rule(self, merged_key):
+        """Just merge all of them (remove duplicates)"""
+        order = CONFIG_RULES.get(merged_key)["inclusive"]
+        merged_value = []
+        for conf in order:
+            value = self.merged_conf[conf].get(merged_key)
+            if value:
+                merged_value.extend(value)
+
+        return merged_value
+
+    def apply_priority_rule(self, merged_key):
+        """
+        Cascade/iterate conf by conf using defined order
+        If undefined at the end, return None
+        """
+        order = CONFIG_RULES.get(merged_key)["priority"]
+        for conf in order:
+            value = self.merged_conf[conf].get(merged_key)
+            if value is not None:
+                return value
+        return None
```

Comment:
We should probably log warning here for now (not sure how we will handle priority mode with env variables) and then return validation error when variable is undefined in final.